haystack/core/super_component/super_component.py

```diff
@@ -0,0 +1,354 @@
+    ) -> None:
+        """
+        Creates a SuperComponent with optional input and output mappings.
+
+        :param pipeline: The pipeline instance to be wrapped
+        :param input_mapping: A dictionary mapping component input names to pipeline input socket paths.
+            If not provided, a default input mapping will be created based on all pipeline inputs.
+        :param output_mapping: A dictionary mapping pipeline output socket paths to component output names.
+            If not provided, a default output mapping will be created based on all pipeline outputs.
+        :raises InvalidMappingError: Raised if any mapping is invalid or type conflicts occur
+        :raises ValueError: Raised if no pipeline is provided
+        """
+        if pipeline is None:
+            raise ValueError("Pipeline must be provided to SuperComponent.")
+
+        self.pipeline: Pipeline = pipeline
+        self._warmed_up = False
+
+        # Determine input types based on pipeline and mapping
+        pipeline_inputs = self.pipeline.inputs()
+        resolved_input_mapping = (
+            input_mapping if input_mapping is not None else self._create_input_mapping(pipeline_inputs)
+        )
+        self._validate_input_mapping(pipeline_inputs, resolved_input_mapping)
+        input_types = self._resolve_input_types_from_mapping(pipeline_inputs, resolved_input_mapping)
+        # Set input types on the component
+        for input_name, info in input_types.items():
+            component.set_input_type(self, name=input_name, **info)
+
+        self.input_mapping: Dict[str, List[str]] = resolved_input_mapping
+        self._original_input_mapping = input_mapping
+
+        # Set output types based on pipeline and mapping
+        pipeline_outputs = self.pipeline.outputs()
+        resolved_output_mapping = (
+            output_mapping if output_mapping is not None else self._create_output_mapping(pipeline_outputs)
+        )
+        self._validate_output_mapping(pipeline_outputs, resolved_output_mapping)
+        output_types = self._resolve_output_types_from_mapping(pipeline_outputs, resolved_output_mapping)
+        # Set output types on the component
+        component.set_output_types(self, **output_types)
+        self.output_mapping: Dict[str, str] = resolved_output_mapping
+        self._original_output_mapping = output_mapping
+
+    def warm_up(self) -> None:
+        """
+        Warms up the SuperComponent by warming up the wrapped pipeline.
+        """
+        if not self._warmed_up:
+            self.pipeline.warm_up()
+            self._warmed_up = True
+
+    def to_dict(self) -> Dict[str, Any]:
+        """
+        Serializes the SuperComponent into a dictionary.
+
+        Must be overwritten for prebuilt super components that inherit from SuperComponent.
+
+        :returns:
+            Dictionary with serialized data.
+        """
+        return self._to_super_component_dict()
```

Comment:
@julian-risch This is why pre-built SuperComponents currently need to always create their own `to_dict` method.

Reply:
Thanks! Let's discuss if/how we can change that in a call.